nautobot/extras/managers.py

```diff
@@ -0,0 +1,30 @@
+from celery import states
+from django.utils import timezone
+from django_celery_results.managers import TaskResultManager, transaction_retry
+
+from nautobot.core.models.querysets import RestrictedQuerySet
+
+
+# This subclass is a hack. We'll fix it in post.
+class JobResultManager(RestrictedQuerySet.as_manager().__class__, TaskResultManager):
+    @transaction_retry(max_retries=2)
+    def store_result(self, *args, **kwargs):
+        """
+        Overload default to manage custom behaviors for Nautobot integration.
+        """
+        obj = super().store_result(*args, **kwargs)
+
+        # Make sure `date_done` is allowed to stay null until the task reacheas
+        # a ready state.
```

Comment:
What's the default behavior of django-celery-results here that we have to override??

Reply:
Default behavior is having this field as a `DateField(auto_now=True)` which just automatically updates the `date_done` field on every state transition. This is different than Celery's default behavior (and the current behavior of Nautobot) to keep it null until there is a state transition to a ready state (e.g. `SUCCESS`, `REVOKED`, `FAILURE`).

So I made it to align with expectations that `date_done` will remain null until there is a ready state, just how it works in Celery core. The default Celery backend does this. The DCR one does not.

Reply:
Would it be more desirable to document that `date_done` behaves more like `last_updated` and that + status would let you know if it's date done or not?

Would we want to change this in the future?